sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -69,6 +71,12 @@ public void reset() {
       putNotNulls(0, capacity);
       numNulls = 0;
     }
+
+    if (hugeVectorThreshold > 0 && capacity > hugeVectorThreshold) {
```

Comment:
shouldn't this be `hugeVectorThreshold > -1`?

Reply:
if hugeVectorThreshold == 0 or hugeVectorThreshold is a small value, the ColumnVector will always releaseMemory() and reserve new memory, this may be slower than before.

Reply:
I know, but according to the doc and impl, this should be > -1, right?

Reply:
Yes, the doc and the code doesn't matched, Sorry.

Reply:
Can you send a followup？

Reply:
Sorry for the later reply, filed a followup PR: https://github.com/apache/spark/pull/47988/files